tune-threads/tune-threads.lisp

```diff
@@ -0,0 +1,92 @@
+;;;; tune-threads/tune-threads.lisp
+;;;;
+;;;; Author: John Lapeyre
+
+(in-package #:qvm-tune-threads)
+
+(defun reset-qvm (q)
+  "Reset the quantum and classical states and program counter of Q."
+  (qvm::reset-quantum-state q)
+  (qvm::reset-classical-memory q)
+  (setf (qvm::pc q) 0))
+
+(defun run-and-reset-qvm (q)
+  "Run the qvm Q, then reset its state."
+  (let ((qvm:*transition-verbose* nil)
+        (qvm::*qubits-required-for-parallelization* 1)); for testing, always use parallelization
+    (qvm:run q)
+    (reset-qvm q)))
+
+(defun time-quil-prog-raw (q &key (num-trials 1) (num-threads 4))
+  "Time the quil program in qvm Q and return the number of seconds."
+  (qvm::prepare-for-parallelization num-threads)
+  (simple-time num-trials (run-and-reset-qvm q)))
+
+(defun time-quil-prog (q &key (num-trials 1) (num-threads 4))
+  "Time the quil program in qvm Q and return the number of seconds per thread per trial."
+  (/ (time-quil-prog-raw q :num-trials num-trials :num-threads num-threads) (* num-threads num-trials)))
+
+(defun find-num-trials (q num-threads time-limit &optional (hint 0))
+  "Choose the number of trials for the qvm/program Q that take around TIME-LIMIT seconds to complete."
+  (if (= hint 1)
+      1
```

Comment:
I still find this `(= hint 1)` special-casing a little mysterious, and am not convinced it wouldn't be better handled at the call site of `find-num-trials`, but I defer to you here since you understand what's going on much better than me.

At a minimum though, it would nice to add a hint (see what I did there) to the documentation string explaining it.